src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -142,10 +146,18 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
         // If the NodeName == "master", we don't set _dd.hostname. It will be overridden by the Datadog Agent. (Traces are only available using Datadog Agent)
         if(!"master".equalsIgnoreCase(nodeName)){
             final String workerHostname = getNodeHostname(run, updatedBuildData);
-            // If the worker hostname is equals to controller hostname but the node name is not "master"
-            // then we could not detect the worker hostname properly. We set _dd.hostname to 'none' explicitly.
+            // If the worker hostname is equals to controller hostname but the node name is not master/built-in then we
+            // could not detect the worker hostname properly. Check if it's set in the environment, otherwise set to none.
             if(buildData.getHostname("").equalsIgnoreCase(workerHostname)) {
-                buildSpan.putMeta(CITags._DD_HOSTNAME, HOSTNAME_NONE);
+                String envHostnameOrNone = HOSTNAME_NONE;
+                try {
+                    Map<String, String> env = run.getEnvironment(new LogTaskListener(logger, Level.INFO));
```

Comment:
This is trick to have the env from the worker? I didn't have this `LogTaskListener` class in my radar.